### PR TITLE
A slew of bugfixes for toddg

### PR DESCRIPTION
**Pull request by rocktavious**
## Issues
There are a number of poorly handled edgecases in the code and this should fix them.

## Changelog

- [X] List your changes here
- [X] Make a `changie` entry

## Tophatting

I used the following terraform configuration

```tf
data "opslevel_tier" "tier_four" {
  filter {
    field = "alias"
    value = "tier_4"
  }
}

data "opslevel_lifecycle" "generally_available" {
  filter {
    field = "alias"
    value = "generally_available"
  }
}

resource "opslevel_team" "terraform_parent_team" {
  name             = "foo"
  responsibilities = "Responsible for foo frontend and backend"
  aliases = ["foo_other"]
}

resource "opslevel_team" "terraform_team" {
  name             = "foo"
  responsibilities = "Responsible for foo frontend and backend"
  aliases = ["terraform_parent_team"]
  parent = opslevel_team.terraform_parent_team.aliases[0]
}

resource "opslevel_service" "terraform_service" {
  name = "terraform-service"
  description = "managed by terraform"
  product = "Product C"
  framework = "Lynx"
  language = "TypeScript"
  lifecycle_alias = data.opslevel_lifecycle.generally_available.alias
  tier_alias = data.opslevel_tier.tier_four.alias
  owner = opslevel_team.terraform_team.aliases[0]

  //aliases = ["terraform-service"]
  tags = ["terraform:true"]
}
```

I tested creating just the teams, modifying the aliases and removing the aliases and everything works as expected and i also tested that after each apply another apply would result in no changes.

### CODE REVIEW

File: Taskfile.yml [hunk 1]
```diff
@@ -22,20 +22,12 @@ tasks:
...
     cmds:
       - go build -o {{.BINARY}} || exit 1
       - chmod +x {{.BINARY}}
-      - mkdir -p {{.BINARY_PATH_BEFORE_VERSION}}/{{.VERSION_FROM_BINARY}}/{{.OS_ARCH}}
-      - mv {{.BINARY}} {{.BINARY_PATH_BEFORE_VERSION}}/{{.VERSION_FROM_BINARY}}/{{.OS_ARCH}}
-      - cmd: echo "Built terraform provider at:"
-        silent: true
-      - cmd: echo "{{.ROOT_DIR}}/{{.BINARY_PATH_BEFORE_VERSION}}/{{.VERSION_FROM_BINARY}}/{{.OS_ARCH}}/{{.BINARY}}"
-        silent: true
+      - mkdir -p {{.LOCATION}} || teue
```
Comment (davidbloss) [inline Taskfile.yml:29]: typo: `teue` --> `true`

File: workspace/Taskfile.yml [hunk 2]
```diff
@@ -30,7 +30,7 @@ tasks:
       - './make_backend_tf.sh'
       - touch main.tf
       - task: terraform-command
-        vars: { TF_COMMAND: "init -upgrade", TF_CMD_DIR: "{{.TASKFILE_DIR}}" }
+        vars: { TF_COMMAND: "init --upgrade", TF_CMD_DIR: "{{.TASKFILE_DIR}}" }
```
Comment (davidbloss) [inline workspace/Taskfile.yml:33]: I'm curious about this change. `terraform init --help` lists `-upgrade` as an option.

File: Taskfile.yml [hunk 3]
```diff
@@ -22,20 +22,12 @@ tasks:
     platforms: [darwin]
     vars:
       BINARY: terraform-provider-opslevel_{{.VERSION_DYNAMIC}}
-      BINARY_PATH_BEFORE_VERSION: "${HOME}/.terraform.d/plugins/registry.terraform.io/opslevel/opslevel"
-      OS_ARCH:
-        sh: echo $(go env GOOS)_$(go env GOARCH)
-      VERSION_FROM_BINARY:
-        sh: echo {{.BINARY}} | cut -d '_' -f 2
+      LOCATION: "${HOME}/.terraform.d/plugins/registry.terraform.io/opslevel/opslevel/{{.VERSION_DYNAMIC}}/{{OS}}_{{ARCH}}"
```
Comment (davidbloss) [inline Taskfile.yml:25]: Nice! I just found [the docs](https://taskfile.dev/usage/#os-specific-taskfiles) that show {{OS}} as an option

File: Taskfile.yml (hunk 1 above)
Comment (rocktavious) [reply Taskfile.yml:29]: @davidbloss - this was failing on my machine without the `|| true` part but i'll try again

File: workspace/Taskfile.yml (hunk 2 above)
Comment (rocktavious) [reply workspace/Taskfile.yml:33]: i'm not sure why i did this.  Very strange.  I don't even recall making this modification

File: Taskfile.yml (hunk 1 above)
Comment (rocktavious) [reply Taskfile.yml:29]: ok this doesn't matter - i swear at one point it was failing for me but could have just been some of the other changes i was making to the path

**Review by davidbloss — APPROVED** [review]
LGTM 🎉

**Pull request merged**